### PR TITLE
Add mouse events

### PR DESCRIPTION
**Pull request by panicbit**

### CODE REVIEW

File: src/rustbox.rs [hunk 1]
```diff
@@ -24,27 +24,33 @@ use libc::c_int;
...
-pub enum InputMode {
-    Current = 0x00,
-
-    /// When ESC sequence is in the buffer and it doesn't match any known
-    /// ESC sequence => ESC means TB_KEY_ESC
-    Esc     = 0x01,
-    /// When ESC sequence is in the buffer and it doesn't match any known
-    /// sequence => ESC enables TB_MOD_ALT modifier for the next keyboard event.
-    Alt     = 0x02,
+bitflags! {
```
Comment (gchp) [inline src/rustbox.rs]: Curious what the reasoning is behind this change?

Comment (panicbit) [inline src/rustbox.rs]: You need to be able to compose these flags as using only `0x04` when setting the input mode makes termbox crash [here with an assertion](https://github.com/nsf/termbox/blob/7c154d98a7d9207d768ee0a8e519ede74c0105cf/src/input.inl#L100) when terminating.

Comment (gchp) [inline src/rustbox.rs]: Ok, cool. Just hit that assertion, prob something that should be fixed in termbox, but anyways.

**Comment by gchp** [comment]
Thanks for this!

**Comment by panicbit** [comment]
After a certain point x-positions of mouse events are reported by termbox as negative. Not sure if this is a bug in the ffi or in termbox itself.

**Comment by gchp** [comment]
Just published `0.6.0` with the mouse support. Will look into that now.

**Comment by panicbit** [comment]
I tested it in C and the issue appears there as well, so it's a bug in termbox.

**Pull request merged**